fdbclient/include/fdbclient/CommitTransaction.h

```diff
@@ -111,23 +117,93 @@ struct MutationRef {
 	}
 
 	std::string toString() const {
-		return format("code: %s param1: %s param2: %s",
+		std::string checksumStr;
+		if (checksum.present()) {
+			checksumStr = format("checksum: %s ", std::to_string(checksum.get()).c_str());
+		}
+		return format("%scode: %s param1: %s param2: %s",
 		              type < MutationRef::MAX_ATOMIC_OP ? typeString[(int)type] : "Unset",
+		              checksumStr.c_str(),
 		              printable(param1).c_str(),
 		              printable(param2).c_str());
 	}
 
+	uint8_t typeWithChecksum() const { return this->type | CHECKSUM_FLAG_MASK; }
+	void removeChecksum() {
+		if (!withChecksum()) {
+			return;
+		}
+		ASSERT(param2.size() >= 4);
+		type &= ~CHECKSUM_FLAG_MASK;
+		param2 = param2.substr(0, param2.size() - 4);
+	}
+	bool withChecksum() const { return this->type & CHECKSUM_FLAG_MASK; }
+
 	bool isAtomicOp() const { return (ATOMIC_MASK & (1 << type)) != 0; }
 	bool isValid() const { return type < MAX_ATOMIC_OP; }
```

Comment:
Are isAtomicOp() and isValid() correct for the case where type is masked by CHECKSUM_FLAG_MASK?